### ISSUE TITLE
batch mode torch.cholesky does not raise RuntimeError on GPU if not pos. def.

### ISSUE THREAD

**Issue by Balandat** [pytorch/pytorch#16780]
## 🐛 Bug

When trying to perform batch Cholesky decomposition of a singular matrix on the GPU, a tensor with NaNs is returned instead of raising a `RuntimeError` as on the CPU.

## To Reproduce

```
torch.cholesky(torch.ones(2, 2))  # -> RuntimeError

torch.cholesky(torch.ones(1, 2, 2))  # -> RuntimeError

torch.cholesky(torch.ones(2, 2, device="cuda"))  # -> RuntimeError

torch.cholesky(torch.ones(1, 2, 2, device="cuda")) # ->
   tensor([[[1., 0.], [1., nan]]], device='cuda:0')
```

## Expected behavior

Behavior should be consistent between CPU and GPU, i.e., a `RuntimeError` should be raised in the last call above.

## Environment

This is on master (fbcode).

**Comment by vishwakftw** [comment]
I will take a look at this. Thank you for reporting.

**Comment by vishwakftw** [comment]
This seems to be a MAGMA issue. I've opened a discussion on the MAGMA forums [here](https://icl.cs.utk.edu/magma/forum/viewtopic.php?f=2&t=2361&sid=d93add864c3744927d3a979423789bf7).

There was a very similar issue with `getrf` (the driver for LU decomposition) which was fixed in MAGMA v2.5.0.

**Comment by vishwakftw** [comment]
I have got the issued fixed the MAGMA team. The fix should be available in their recent release: v2.5.1-alpha1.

cc: @soumith

**Comment by Balandat** [comment]
Can we update to this new release soon? This kind of annoying to have to work around, I currently do the following: 

    L = torch.cholesky(A, upper=upper, out=out)
    # TODO: Remove once fixed in pytorch (#16780)
    if A.dim() > 2 and A.is_cuda:
        if torch.isnan(L if out is None else out).any():
            raise RuntimeError

**Comment by vishwakftw** [comment]
cc: @soumith is this a good time to release the 2.5.1alpha binaries for MAGMA?

**Comment by soumith** [comment]
@vishwakftw yes i guess it's time :-/

**Comment by vishwakftw** [comment]
@Balandat, the latest MAGMA binaries are available on Anaconda, thanks to @soumith.

The fix to this issue is available in this binary, and hence marking this issue as resolved. Please re-open if you think this is not the case.
